### PR TITLE
Item class and person data

### PR DESCRIPTION
**Pull request by waltjones**
## Description of the change

The motivation for this PR is to add person data (current user) as supported in other Rollbar SDKs. This PR also includes refactoring to use an `Item` class to manage setting up the occurrence, as is done in most or all of the other SDKs.

The original, minimal implementation of this SDK didn't use an Item class, and any data, flags or modifiers of the payload needed to be passed through as method arguments. As more optional data elements have been added, e.g. custom data, telemetry, and now person data, this arrangement has been stretched to its limit and keeping everything in an Item class is a much needed simplification. 

This is a relatively large PR, but is cleanly divided into commits that can each be reviewed separately:
- [x] refactor to use Item class, and set notfier.diagnostic key in payload
- [x] extract user id from Flow and Exception emails
- [x] add person data to payload, either via item.personId or current user
- [x] add config settings to enable/disable person id, username, email

**Person data in Salesforce:**
Salesforce allows using the `UserInfo` service to get the current user in most contexts. This does not work when processing unhandled Exceptions and Flow errors, as these run outside the user's context. In those cases, the user id is extracted from the error data and used to perform a DML lookup (essentially a database query).

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Fixes ch73809

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: force-app/main/default/classes/Notifier.cls [hunk 1]
```diff
@@ -9,19 +9,10 @@ public with sharing class Notifier
...
-    public HttpResponse log(String level, String message, Map<String, Object> custom, List<Telemetry> telemetry, SendMethod method)
-    {
-        return send(dataBuilder.buildPayload(level, message, custom, telemetry), method);
-    }
-
-    public HttpResponse log(String level, Exception exc, Map<String, Object> custom, SendMethod method)
-    {
-        return send(dataBuilder.buildPayload(level, exc, custom), method);
-    }
+    public HttpResponse log(String level, Item item, SendMethod method){
```
Comment (ajtran) [inline force-app/main/default/classes/Notifier.cls:12]: 👍

File: force-app/main/default/classes/RollbarExceptionEmailHandler.cls [hunk 2]
```diff
@@ -47,17 +47,20 @@ global class RollbarExceptionEmailHandler implements Messaging.InboundEmailHandl
...
             if (!settings.ReportFlowErrors__c) { return; }
 
             emailBody = email.htmlBody;
-            List<Telemetry> telemetry = FlowEmailParser.parseTelemetry(emailBody);
+            FlowEmailParser flowParser = new FlowEmailParser(emailBody);
+            List<Telemetry> telemetry = flowParser.parseTelemetry();
 
             Item item = new Item(email.subject);
             item.isUncaught = true;
+            item.personId = flowParser.currentUserId;
```
Comment (ajtran) [inline force-app/main/default/classes/RollbarExceptionEmailHandler.cls:63]: 👍

File: force-app/main/default/classes/DataBuilder.cls [hunk 3]
```diff
@@ -222,5 +225,44 @@ public with sharing class DataBuilder {
...
+
+        return buildCurrentPersonData();
+    }
+
+    private Map<String, Object> buildCurrentPersonData() {
+        Map<String, Object> data = new Map<String, Object>();
+
+        data.put('id', UserInfo.getUserId());
+        data.put('username', UserInfo.getUserName());
+        data.put('email', UserInfo.getUserEmail());
```
Comment (ajtran) [inline force-app/main/default/classes/DataBuilder.cls]: Where is `UserInfo` set or defined?

File: force-app/main/default/classes/DataBuilder.cls [hunk 4]
```diff
@@ -30,6 +30,9 @@ public with sharing class DataBuilder {
             data.put('custom', item.custom);
         }
 
+        // TODO: if config.personData
+        data.put('person', buildPersonData(item));
```
Comment (ajtran) [inline force-app/main/default/classes/DataBuilder.cls]: 👍

**Review by ajtran — APPROVED** [review]
Had a question for my own curiosity, but looks good! Breaking it down into commits was very helpful, great iterative work!!

File: force-app/main/default/classes/DataBuilder.cls (hunk 3 above)
Comment (waltjones) [reply force-app/main/default/classes/DataBuilder.cls]: It's a system provided service. https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_userinfo.htm

**Comment by waltjones** [comment]
@ajtran Thank you!

**Review by nathanvoller — COMMENTED** [review]
Fantastic job on this Walt! So pleased to see the Salesforce SDK get support for person data 🤩

Quick question around API versions - are you planning on keeping these up to date or at least consistent?

**Comment by waltjones** [comment]
It's great to hear people are excited about it. Thanks for the comment.

Sorry I'm not sure which API you're discussing. Do you mean the version in the class metadata? `<apiVersion>47.0</apiVersion>`

**Comment by nathanvoller** [comment]
Yes exactly right. From what I can see in the source, the versions range from 42-47 currently. It's certainly not a requirement to keep up to date but version 50 has a couple of new features that might be of interest, namely [Apex Runtime Context](https://help.salesforce.com/articleView?id=release-notes.rn_apex_Runtime_Detection.htm&type=5&release=228) with the `Request.getQuiddity()` method.

File: force-app/main/default/classes/DataBuilder.cls (hunk 3 above)
Comment (ajtran) [reply force-app/main/default/classes/DataBuilder.cls]: I see! Thanks!

**Comment by waltjones** [comment]
Thanks for mentioning Apex Runtime Context. I'll check it out.

**Pull request merged**